storage/connectors/bigquery/src/main/java/feast/storage/connectors/bigquery/compression/FeatureRowsBatch.java

```diff
@@ -0,0 +1,228 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.storage.connectors.bigquery.compression;
+
+import static feast.proto.types.ValueProto.Value.ValCase.*;
+import static feast.storage.connectors.bigquery.common.TypeUtil.*;
+
+import feast.proto.types.FeatureRowProto;
+import feast.proto.types.FieldProto;
+import feast.proto.types.ValueProto;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.beam.sdk.coders.*;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.values.Row;
+
+/**
+ * Based on list of featureRows this class infers common schema (with all found features) and then
+ * transpose list of rows into list of values format (column-oriented).
+ *
+ * <p>getFeatureRows provides reverse tranformation
+ */
+public class FeatureRowsBatch implements Serializable {
+  private final Schema schema;
+  private String featureSetReference;
+  private List<Object> values = new ArrayList<>();
+
+  public static Map<ValueProto.Value.ValCase, Schema.FieldType> protoToSchemaTypes =
+      new HashMap<>();
+  public static Map<Schema.FieldType, ValueProto.Value.ValCase> schemaToProtoTypes;
+  public static Map<String, Object> defaultValues = new HashMap<>();
+
+  static {
+    protoToSchemaTypes.put(BYTES_VAL, Schema.FieldType.BYTES);
+    protoToSchemaTypes.put(STRING_VAL, Schema.FieldType.STRING);
+    protoToSchemaTypes.put(INT32_VAL, Schema.FieldType.INT32);
+    protoToSchemaTypes.put(INT64_VAL, Schema.FieldType.INT64);
+    protoToSchemaTypes.put(DOUBLE_VAL, Schema.FieldType.DOUBLE);
+    protoToSchemaTypes.put(FLOAT_VAL, Schema.FieldType.FLOAT);
+    protoToSchemaTypes.put(BOOL_VAL, Schema.FieldType.BOOLEAN);
+    protoToSchemaTypes.put(BYTES_LIST_VAL, Schema.FieldType.array(Schema.FieldType.BYTES));
+    protoToSchemaTypes.put(STRING_LIST_VAL, Schema.FieldType.array(Schema.FieldType.STRING));
+    protoToSchemaTypes.put(INT32_LIST_VAL, Schema.FieldType.array(Schema.FieldType.INT32));
+    protoToSchemaTypes.put(INT64_LIST_VAL, Schema.FieldType.array(Schema.FieldType.INT64));
+    protoToSchemaTypes.put(FLOAT_LIST_VAL, Schema.FieldType.array(Schema.FieldType.FLOAT));
+    protoToSchemaTypes.put(BOOL_LIST_VAL, Schema.FieldType.array(Schema.FieldType.BOOLEAN));
+    protoToSchemaTypes.put(DOUBLE_LIST_VAL, Schema.FieldType.array(Schema.FieldType.DOUBLE));
+
+    schemaToProtoTypes =
+        protoToSchemaTypes.entrySet().stream()
+            .collect(Collectors.toMap(Map.Entry::getValue, Map.Entry::getKey));
+  }
+
+  public FeatureRowsBatch(Iterable<FeatureRowProto.FeatureRow> featureRows) {
+    schema = inferCommonSchema(featureRows);
+    initValues();
```

Comment:
Can we use `this.`, small nitpick.

Reply:
Also, is there a reason that functionality is not in the constructor?

Reply:
it was just too many lines of code, decided to split them into methods